### ISSUE TITLE
Can't remove rate limiting periods

### ISSUE THREAD

**Issue by rocketspacer** [Kong/kong#3940]
### Summary

Calling Admin API to remove a rate-limiting period results in HTTP 500
```
{
    "message": "An unexpected error occurred"
}
```

### Steps To Reproduce

1. Configure rate-limiting with hour=30
2. Configure rate-limiting with minute=10
3. Generate some requests to use the rate-limit
4. Configure rate-limting with minute=null

```
curl -X PATCH \
  http://localhost:8001/plugins/9780e658-f1ff-4f59-9b67-7e95eb0dc90e \
  -H 'Content-Type: application/json' \
  -d '{
	"config": {
		"hour": "30",
		"minute": null
	}
}'
```

### Additional Details & Logs

- Kong version: 0.14.1
- Kong error logs
```
2018/10/31 11:37:31 [error] 354#0: *3056 [lua] init.lua:133: handle_error(): /usr/local/share/lua/5.1/lapis/application.lua:397: /kong-plugins/kong/plugins/x-quota/schema.lua:32: attempt to compare number with userdata
stack traceback:
        /kong-plugins/kong/plugins/x-quota/schema.lua: in function 'self_check'
        /usr/local/share/lua/5.1/kong/dao/schemas/plugins.lua:98: in function 'self_check'
        /usr/local/share/lua/5.1/kong/dao/schemas_validation.lua:289: in function 'validate'
        /usr/local/share/lua/5.1/kong/dao/model_factory.lua:27: in function 'validate'
        /usr/local/share/lua/5.1/kong/dao/dao.lua:301: in function 'update'
        /usr/local/share/lua/5.1/kong/api/crud_helpers.lua:226: in function 'patch'
        /usr/local/share/lua/5.1/kong/api/routes/plugins.lua:70: in function </usr/local/share/lua/5.1/kong/api/routes/plugins.lua:69>

stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/lapis/application.lua:397: in function 'handler'
        /usr/local/share/lua/5.1/lapis/application.lua:130: in function 'resolve'
        /usr/local/share/lua/5.1/lapis/application.lua:161: in function </usr/local/share/lua/5.1/lapis/application.lua:159>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/lapis/application.lua:159: in function 'dispatch'
        /usr/local/share/lua/5.1/lapis/nginx.lua:215: in function 'serve_admin_api'
        content_by_lua(nginx-kong.conf:154):2: in function <content_by_lua(nginx-kong.conf:154):1>, client: 172.31.0.1, server: kong_admin, request: "PATCH /plugins/9780e658-f1ff-4f59-9b67-7e95eb0dc90e HTTP/1.1", host: "localhost:8001"
172.31.0.1 - - [31/Oct/2018:11:37:31 +0000] "PATCH /plugins/9780e658-f1ff-4f59-9b67-7e95eb0dc90e HTTP/1.1" 500 43 "-" "PostmanRuntime/7.3.0"
```
- Kong configuration
```
{
  "plugins": {
    "enabled_in_cluster": ["grabid_hmac", "rate-limiting", "syslog", "grabid", "datadog"],
    "available_on_server": {
      "correlation-id": true,
      "pre-function": true,
      "cors": true,
      "grabid_hmac": true,
      "rate-limiting": true,
      "loggly": true,
      "hmac-auth": true,
      "zipkin": true,
      "request-size-limiting": true,
      "azure-functions": true,
      "request-transformer": true,
      "grabid": true,
      "oauth2": true,
      "response-transformer": true,
      "ip-restriction": true,
      "statsd": true,
      "jwt": true,
      "basic-auth": true,
      "key-auth": true,
      "http-log": true,
      "datadog": true,
      "tcp-log": true,
      "post-function": true,
      "rewrite_uri": true,
      "acl": true,
      "prometheus": true,
      "syslog": true,
      "response-ratelimiting": true,
      "udp-log": true,
      "file-log": true,
      "aws-lambda": true,
      "bot-detection": true,
      "ldap-auth": true,
      "request-termination": true
    }
  },
  "tagline": "Welcome to kong",
  "configuration": {
    "plugins": ["bundled", "grabid", "datadog", "grabid_hmac", "rewrite_uri", "statsd"],
    "admin_ssl_enabled": false,
    "lua_ssl_verify_depth": 1,
    "trusted_ips": ["10.10.0.0\/16"],
    "prefix": "\/usr\/local\/kong",
    "loaded_plugins": {
      "correlation-id": true,
      "pre-function": true,
      "cors": true,
      "grabid_hmac": true,
      "rate-limiting": true,
      "loggly": true,
      "hmac-auth": true,
      "zipkin": true,
      "bot-detection": true,
      "azure-functions": true,
      "request-transformer": true,
      "grabid": true,
      "oauth2": true,
      "response-transformer": true,
      "ip-restriction": true,
      "statsd": true,
      "jwt": true,
      "basic-auth": true,
      "key-auth": true,
      "http-log": true,
      "datadog": true,
      "tcp-log": true,
      "post-function": true,
      "rewrite_uri": true,
      "acl": true,
      "syslog": true,
      "ldap-auth": true,
      "response-ratelimiting": true,
      "udp-log": true,
      "file-log": true,
      "aws-lambda": true,
      "request-size-limiting": true,
      "prometheus": true,
      "request-termination": true
    },
    "cassandra_username": "kong",
    "admin_ssl_cert_csr_default": "\/usr\/local\/kong\/ssl\/admin-kong-default.csr",
    "dns_resolver": {},
    "pg_user": "kong",
    "pg_password": "******",
    "cassandra_data_centers": ["dc1:2", "dc2:3"],
    "nginx_admin_directives": {},
    "custom_plugins": {},
    "pg_host": "db-partner-gw.stg-myteksi.com",
    "nginx_acc_logs": "\/usr\/local\/kong\/logs\/access.log",
    "proxy_listen": ["0.0.0.0:8000"],
    "client_ssl_cert_default": "\/usr\/local\/kong\/ssl\/kong-default.crt",
    "ssl_cert_csr_default": "\/usr\/local\/kong\/ssl\/kong-default.csr",
    "dns_no_sync": false,
    "db_update_propagation": 0,
    "nginx_err_logs": "\/usr\/local\/kong\/logs\/error.log",
    "cassandra_port": 9042,
    "dns_order": ["LAST", "SRV", "A", "CNAME"],
    "dns_error_ttl": 1,
    "headers": ["off"],
    "dns_stale_ttl": 5,
    "nginx_optimizations": true,
    "database": "postgres",
    "pg_database": "kong",
    "nginx_worker_processes": "auto",
    "lua_package_cpath": "",
    "admin_acc_logs": "\/usr\/local\/kong\/logs\/admin_access.log",
    "lua_package_path": "\/home\/grabtaxi\/apps\/partner-gw\/latest\/?.lua;",
    "nginx_pid": "\/usr\/local\/kong\/pids\/nginx.pid",
    "upstream_keepalive": 60,
    "admin_access_log": "\/var\/log\/grabtaxi\/partner-gw_admin.log",
    "client_ssl_cert_csr_default": "\/usr\/local\/kong\/ssl\/kong-default.csr",
    "proxy_listeners": [{
      "ssl": false,
      "ip": "0.0.0.0",
      "proxy_protocol": false,
      "port": 8000,
      "http2": false,
      "listener": "0.0.0.0:8000"
    }],
    "proxy_ssl_enabled": false,
    "cassandra_contact_points": ["127.0.0.1"],
    "enabled_headers": {
      "latency_tokens": false,
      "X-Kong-Proxy-Latency": false,
      "Via": false,
      "server_tokens": false,
      "Server": false,
      "X-Kong-Upstream-Latency": false,
      "X-Kong-Upstream-Status": false
    },
    "ssl_ciphers": "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256",
    "cassandra_ssl": false,
    "db_resurrect_ttl": 10,
    "ssl_cert_key_default": "\/usr\/local\/kong\/ssl\/kong-default.key",
    "cassandra_consistency": "ONE",
    "db_cache_ttl": 0,
    "admin_error_log": "\/var\/log\/grabtaxi\/partner-gw_admin.err",
    "pg_ssl_verify": false,
    "dns_not_found_ttl": 10,
    "pg_ssl": false,
    "client_max_body_size": "0",
    "client_ssl": false,
    "cassandra_repl_strategy": "SimpleStrategy",
    "ssl_cipher_suite": "modern",
    "lua_socket_pool_size": 30,
    "db_update_frequency": 5,
    "nginx_http_directives": [{
      "value": "prometheus_metrics 5m",
      "name": "lua_shared_dict"
    }],
    "kong_env": "\/usr\/local\/kong\/.kong_env",
    "cassandra_schema_consensus_timeout": 10000,
    "dns_hostsfile": "\/etc\/hosts",
    "admin_listeners": [{
      "ssl": false,
      "ip": "0.0.0.0",
      "proxy_protocol": false,
      "port": 8001,
      "http2": false,
      "listener": "0.0.0.0:8001"
    }],
    "cassandra_lb_policy": "RoundRobin",
    "error_default_type": "text\/plain",
    "real_ip_header": "X-Real-IP",
    "admin_ssl_cert_key_default": "\/usr\/local\/kong\/ssl\/admin-kong-default.key",
    "cassandra_ssl_verify": false,
    "proxy_error_log": "\/var\/log\/grabtaxi\/partner-gw.err",
    "nginx_proxy_directives": {},
    "real_ip_recursive": "off",
    "cassandra_repl_factor": 1,
    "client_ssl_cert_key_default": "\/usr\/local\/kong\/ssl\/kong-default.key",
    "nginx_daemon": "on",
    "anonymous_reports": false,
    "cassandra_timeout": 5000,
    "proxy_access_log": "\/var\/log\/grabtaxi\/partner-gw.log",
    "pg_port": 5432,
    "nginx_kong_conf": "\/usr\/local\/kong\/nginx-kong.conf",
    "client_body_buffer_size": "8k",
    "mem_cache_size": "128m",
    "admin_ssl_cert_default": "\/usr\/local\/kong\/ssl\/admin-kong-default.crt",
    "log_level": "notice",
    "cassandra_keyspace": "kong",
    "ssl_cert_default": "\/usr\/local\/kong\/ssl\/kong-default.crt",
    "nginx_conf": "\/usr\/local\/kong\/nginx.conf",
    "admin_listen": ["0.0.0.0:8001"]
  },
  "version": "0.14.1",
  "node_id": "f292d0f3-4098-41eb-a747-5403ce79491e",
  "lua_version": "LuaJIT 2.1.0-beta3",
  "prng_seeds": {
    "pid: 20397": 140101531651,
    "pid: 20400": 202547056184,
    "pid: 20398": 114221159215,
    "pid: 20396": 194176194618,
    "pid: 20401": 601372355827,
    "pid: 20399": 240912495614,
    "pid: 20402": 238148581561,
    "pid: 20395": 164891631926
  },
  "timers": {
    "pending": 3,
    "running": 0
  },
  "hostname": "ip-10-10-0-22"
}
```
- Operating system: Kong alpine Docker image

**Comment by p0pr0ck5** [comment]
I believe this is solved with the release of Kong 1.0. @rocketspacer can you try this with the latest release?

**Comment by p0pr0ck5** [comment]
I'm going to close this, given the lack of response and the staleness of the issue. @rocketspacer or anyone else if you encounter this with Kong 1.0, please feel free to leave a comment here and we can re-open the issue and examine. Thank you!
